event_response_tool/event_response.py

```diff
@@ -232,47 +364,101 @@ def construct_argument_parser():
     parser.add_argument("--username", help="AnalyzeRe Username")
     parser.add_argument("--password", help="Analyzere Password")
 
-    # Required event weight CSV
+    # Event weight CSV
     parser.add_argument(
         "--event_weights_csv",
         help="The path of the CSV file containing events and their "
         "corresponding weights",
+    )
+
+    # Choices for deciding between Mixture distribution and convolution method
+    parser.add_argument(
+        "--method",
+        dest="method",
+        choices=["mixture_distribution", "convolution"],
+        help="The method to use for Event Response analysis",
         required=True,
     )
 
-    # LayerView manipulation arguments
+    # Method 1 - Mixture Distribution
     parser.add_argument(
-        "--layer_views_csv",
-        help="The path of the CSV file containing LayerView UUIDs",
+        "--old_analysis_profile_uuid",
+        help="The UUID of the existing Analysis Profile that needs to be updated",
     )
     parser.add_argument(
-        "--portfolio_view_uuid",
-        help="The UUID of PortfolioView containing the required LayerViews",
+        "--max_trial_per_event",
+        help="Maximum trial per event in the weighted Simulation grid",
+        type=int,
+    )
+    parser.add_argument(
+        "--simulation_description_mixture_distribution",
+        help="Description of the new Simulation grid to be created, if not provided, the default value from config file is used",
     )
+    parser.add_argument(
+        "--analysis_profile_description_mixture_distribution",
+        help="Description of the new Analysis Profile to be created, if not provided, the default value from config file is used",
+    )
+    parser.add_argument(
+        "--simulation_start_date_mixture_distribution",
+        help="Start date of the Simulation, if not provided, the date defaults to original Simgrid start date",
+    )
+
+    # Method 2 - Convolution
+
+    # Flag to decide if a new Analysis Profile is to be created
+    parser.add_argument(
+        "--new_analysis_profile",
+        dest="create_analysis_profile",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--no_new_analysis_profile",
+        dest="create_analysis_profile",
+        action="store_false",
+    )
```

Comment:
Do we need both of these?

Reply:
Yes, this is as same as using the "create_analysis_profile" boolean flag in the notebook. The parameter `--new_analysis_analysis_profile` sets the boolean flag `create_analysis_profile` to True while the `--no_new_analysis_profile` option sets the flag to False (setting True and False is handled by the `action` value in `add_argument`). Argparse doesn't do a great job in parsing boolean values when provided as a direct input via the CLI, so went with this approach.

Reply:
Does using a `default` not allow you to just have one parameter? e.g. `--new_analysis_analysis_profile` sets the boolean flag `create_analysis_profile` to `True` and `default=False`?

Reply:
Agreed! Made the change now